oak-blob-cloud-azure/src/main/java/org/apache/jackrabbit/oak/blob/cloud/azure/blobstorage/AzureBlobStoreBackend.java

```diff
@@ -658,29 +545,23 @@ public List<DataRecord> getAllMetadataRecords(String prefix) {
         try {
             Thread.currentThread().setContextClassLoader(getClass().getClassLoader());
 
-            CloudBlobDirectory metaDir = getAzureContainer().getDirectoryReference(META_DIR_NAME);
-            for (ListBlobItem item : metaDir.listBlobs(prefix)) {
-                if (item instanceof CloudBlob) {
-                    CloudBlob blob = (CloudBlob) item;
-                    blob.downloadAttributes();
-                    records.add(new AzureBlobStoreDataRecord(
-                            this,
-                            azureBlobContainerProvider,
-                            new DataIdentifier(stripMetaKeyPrefix(blob.getName())),
-                            getLastModified(blob),
-                            blob.getProperties().getLength(),
-                            true));
-                }
+            for (BlobItem blobItem : getAzureContainer().listBlobsByHierarchy(META_DIR_NAME)) {
```

Comment:
`listBlobsByHierarchy` is listing only the hierarchy, so 'META/'.
this has to be changed with
```
ListBlobsOptions listBlobsOptions = new ListBlobsOptions();
listBlobsOptions.setPrefix(META_DIR_NAME);
getAzureContainer().listBlobs(listBlobsOptions, null);
```